src/Datadog.Trace.ClrProfiler.Managed/CallTarget/Handlers/IntegrationMapper.cs

```diff
@@ -0,0 +1,714 @@
+            // Log.Information("Generic Types: " + string.Join(", ", callGenericTypes.Select(t => t.FullName)));
+            // Log.Information("Method: " + onMethodBeginMethodInfo);
+            onMethodBeginMethodInfo = onMethodBeginMethodInfo.MakeGenericMethod(callGenericTypes.ToArray());
+            // Log.Information("Method: " + onMethodBeginMethodInfo);
+            ilWriter.EmitCall(OpCodes.Call, onMethodBeginMethodInfo, null);
+            ilWriter.Emit(OpCodes.Ret);
+
+            Log.Debug($"Created SlowBeginMethod Dynamic Method for '{integrationType.FullName}' integration. [Target={targetType.FullName}]");
+            return callMethod;
+        }
+
+        internal static DynamicMethod CreateEndMethodDelegate(Type integrationType, Type targetType)
+        {
+            Log.Debug($"Creating EndMethod Dynamic Method for '{integrationType.FullName}' integration. [Target={targetType.FullName}]");
+            MethodInfo onMethodEndMethodInfo = integrationType.GetMethod(EndAsyncMethodName, BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Static);
+            if (onMethodEndMethodInfo is null)
+            {
+                return null;
+            }
+
+            if (onMethodEndMethodInfo.ReturnType != typeof(CallTargetReturn))
+            {
+                throw new ArgumentException($"The return type of the method: {EndAsyncMethodName} in type: {integrationType.FullName} is not {nameof(CallTargetReturn)}");
+            }
+
+            Type[] genericArgumentsTypes = onMethodEndMethodInfo.GetGenericArguments();
+            if (genericArgumentsTypes.Length != 1)
+            {
+                throw new ArgumentException($"The method: {EndAsyncMethodName} in type: {integrationType.FullName} must have a single generic type for the instance type.");
+            }
+
+            ParameterInfo[] onMethodEndParameters = onMethodEndMethodInfo.GetParameters();
+            if (onMethodEndParameters.Length < 2)
+            {
+                throw new ArgumentException($"The method: {EndAsyncMethodName} with {onMethodEndParameters.Length} paremeters in type: {integrationType.FullName} has less parameters than required.");
+            }
+            else if (onMethodEndParameters.Length > 3)
+            {
+                throw new ArgumentException($"The method: {EndAsyncMethodName} with {onMethodEndParameters.Length} paremeters in type: {integrationType.FullName} has more parameters than required.");
+            }
+
+            if (onMethodEndParameters[0].ParameterType != typeof(Exception) && onMethodEndParameters[1].ParameterType != typeof(Exception))
+            {
+                throw new ArgumentException($"The Exception type parameter of the method: {EndAsyncMethodName} in type: {integrationType.FullName} is missing.");
+            }
+
+            if (onMethodEndParameters[1].ParameterType != typeof(CallTargetState) && onMethodEndParameters[2].ParameterType != typeof(CallTargetState))
```

Comment:
`onMethodEndParameters.Length` may equal 2 at this point, so accessing `onMethodEndParameters[2]` may cause an out of bounds exception

Reply:
fixed in: https://github.com/DataDog/dd-trace-dotnet/pull/1007/commits/04ae61a340872f36f5e93929e69b26305bc991dd